src/main/java/com/mparticle/kits/IterableKit.java

```diff
@@ -35,10 +76,17 @@ public String getName() {
         return null;
     }
 
+    private void initIntegrationAttributes() {
+        HashMap<String, String> integrationAttributes = new HashMap<>();
+        integrationAttributes.put(INTEGRATION_ATTRIBUTE_KIT_VERSION_CODE, String.valueOf(BuildConfig.VERSION_CODE));
+        integrationAttributes.put(INTEGRATION_ATTRIBUTE_SDK_VERSION, com.iterable.iterableapi.BuildConfig.ITERABLE_SDK_VERSION);
+        setIntegrationAttributes(integrationAttributes);
+    }
+
     private void checkForAttribution() {
-        Activity activity = getKitManager().getCurrentActivity().get();
-        if (activity != null) {
-            String currentLink = activity.getIntent().getDataString();
+        WeakReference<Activity> activity = getKitManager().getCurrentActivity();
+        if (activity != null && activity.get() != null) {
+            String currentLink = activity.get().getIntent().getDataString();
```

Comment:
`activity.get()` could end up returning `null` here since you're no longer grabbing a strong reference to the underlying `Activity` object

Reply:
I'm not sure what you mean.. There is a check for `activity.get() != null` right above it. Or do you mean that this variable can be modified from another thread and the operation should be atomic?

Reply:
Or, rather, paused between these two lines and garbage collected?

Reply:
Ok, removed the second call to `get()`.